### PR TITLE
Node3 test setups

### PR DESCRIPTION
**Pull request by luismiramirez**
This PR is a migration of the test setups present in the Node.js integration test suite and the adaptation to make them work as any other test setup present in this repository. The Prisma and Knex ones have been omitted as they only differ from the Express + Database setups in one line that can be easily edited in place.

The rest of the setups are now present in here and edited to comply with the CI tests.

Closes https://github.com/appsignal/appsignal-nodejs/issues/792

### CODE REVIEW

**Review by shairyar — APPROVED** [review]

**Comment by unflxw** [comment]
This doesn't have to be done in this PR, but I think it's worth mentioning: a few months ago I ["fixed" the Node.js 2.x test setups](https://github.com/appsignal/test-setups/pull/74) by making them [check out the `2-4-stable` branch](https://github.com/appsignal/test-setups/pull/74/files#diff-7df915229f542d4d139b8ae6eec7e1cc50c9c873932348f8eb52144fe1fa76e8R8) when they're booted.

This was a quick, hacky fix, and it has ugly consequences -- if you run a 2.x test setup, it will "break" 3.x setups in a way that's kinda hard to figure out. And if you want to run a 2.x test setup with code from a separate branch, it will silently check out the `2-4-stable` branch instead when you run it.

Now that there's separate `nodejs/3.x` and `nodejs/2.x` folders, this gives us the opportunity to clone the integration twice, once for 2.x and once for 3.x, checking out different branches each time:
```
- nodejs/
  - 2.x/
    - integration/ -> `2-4-stable` branch
  - 3.x/
    - integration/ -> `main` branch
```

**Review by unflxw — APPROVED** [review]
Looks good overall! What follows are future to-dos, but not blockers for this PR.

- We should probably not use `appsignal.ts` as the AppSignal initialisation file, instead using an `appsignal.cjs` file that is not affected by the build, like the ones that we tell users to create and that are created by our installer.
- I understand your decision to omit Prisma and Knex, given that we already have a lot of almost identical "Express + WhateverDB" test setups. But I would still like to have a Prisma one, because Prisma is popular, it's a bit elaborate to set up, and their OpenTelemetry support is experimental and could change at any moment. [I've created a separate issue for this.](https://github.com/appsignal/test-setups/issues/101)

File: nodejs/v3/express-mongoose/app/src/appsignal.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { Appsignal } from "@appsignal/nodejs"
+
+export default new Appsignal({
```
Comment (tombruijn) [inline nodejs/v3/express-mongoose/app/src/appsignal.ts:3]:
```suggestion
new Appsignal({
```
Export not needed, right?

File: nodejs/v3/express-postgres/app/src/appsignal.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { Appsignal } from "@appsignal/nodejs"
+
+export default new Appsignal({
```
Comment (tombruijn) [inline nodejs/v3/express-postgres/app/src/appsignal.ts:3]:
```suggestion
new Appsignal({
```

**Review by tombruijn — APPROVED** [review]
I'm not reviewing every line in this PR, but agree with the general ideas. Agree with Noemi says about the appsignal.cjs file.

**Comment by luismiramirez** [comment]
> This was a quick, hacky fix to make the CI green, and it has ugly consequences in practice -- if you run a 2.x test setup, it will "break" 3.x setups in a way that's kinda hard to figure out. And if you want to run a 2.x test setup with code from a separate branch, it will silently check out the 2-4-stable branch instead when you run it.

I think it's a good idea, but is it worth it? I mean, v2 could be considered deprecated, at least in a soft way.

**Pull request merged**